Examples/Detectors/GenericDetector/src/GenericDetector.cpp

```diff
@@ -323,6 +323,427 @@ Gen1GenericDetectorBuilder::buildTrackingGeometry(
+
+  root.addCylinderContainer("Detector", AxisR, [&](auto& detector) {
+    auto beampipeBounds = std::make_unique<Acts::CylinderVolumeBounds>(
+        0_mm, kBeamPipeRadius, kBeamPipeHalfLengthZ);
+    auto beampipe = std::make_unique<Acts::TrackingVolume>(
+        Acts::Transform3::Identity(), std::move(beampipeBounds), "Beampipe");
+
+    detector.addMaterial("BeampipeMaterial", [&](auto& bpMat) {
+      bpMat.configureFace(OuterCylinder,
+                          asHomogeneous("Beam pipe", m_beamPipeMaterial));
+      bpMat.addStaticVolume(std::move(beampipe));
+    });
+
+    buildPixel(detector, gctx);
+
+    if (m_cfg.buildLevel > 1) {
+      buildShortStrip(detector, gctx);
+      buildLongStrip(detector, gctx);
+    }
+  });
+
+  // @TODO: Add Pixel Support Tube
+  // @TODO: Add Short Strip
+  // @TODO: Add Long Strip
+
+  if (graphvizFile) {
+    std::ofstream file(*graphvizFile);
+    root.graphviz(file);
+  }
+
+  BlueprintOptions options;
+  auto trackingGeometry =
+      root.construct(options, gctx, *logger().clone("Blprnt"));
+  return trackingGeometry;
+}
+
+void Gen3GenericDetectorBuilder::buildPixel(
+    Acts::Experimental::BlueprintNode& parent,
+    const Acts::GeometryContext& gctx) {
+  using enum Acts::AxisDirection;
+  using namespace Acts::Experimental;
+  using namespace Acts::UnitLiterals;
+  using enum Acts::CylinderVolumeBounds::Face;
+  using AttachmentStrategy = Acts::VolumeAttachmentStrategy;
+  using ResizeStrategy = Acts::VolumeResizeStrategy;
+
+  ACTS_DEBUG("Building Pixel");
+
+  ProtoLayerCreator pplCreator = createPixelProtoLayerCreator();
+
+  auto& pixel = parent.addCylinderContainer("Pixel", AxisR);
+
+  if (m_cfg.buildLevel > 1) {
+    ACTS_DEBUG("Building PST");
+    auto pstVolume = std::make_unique<Acts::TrackingVolume>(
+        Acts::Transform3::Identity(),
+        std::make_unique<Acts::CylinderVolumeBounds>(
+            kPstRadius - kPstThickness, kPstRadius, kPstHalfLengthZ),
+        "PST");
+
+    pixel.addMaterial("PSTMaterial", [&](auto& pstMat) {
+      pstMat.configureFace(OuterCylinder, asHomogeneous("PST", m_pstMaterial));
+      pstMat.addStaticVolume(std::move(pstVolume));
+    });
+  }
+
+  auto& sensitive = pixel.addCylinderContainer("Pixel", AxisZ);
+  sensitive.addCylinderContainer("Pixel_Barrel", AxisR, [&](auto& barrel) {
+    barrel.setAttachmentStrategy(AttachmentStrategy::Gap);
+    barrel.setResizeStrategies(ResizeStrategy::Expand, ResizeStrategy::Gap);
+
+    auto centralProtoLayerSurfaces = pplCreator.centralProtoLayers(gctx);
+    ACTS_DEBUG("Adding " << centralProtoLayerSurfaces.size()
+                         << " central proto layers to "
+                         << "Pixel_Barrel");
+    for (const auto& [idx, temp] :
+
+         Acts::enumerate(centralProtoLayerSurfaces)) {
+      auto& pl = temp.protoLayer;
+      std::string layerName = "Pixel_Barrel_L" + std::to_string(idx);
+      barrel.addMaterial(layerName + "_Mat", [&](auto& mat) {
```

Comment:
Do you want to have some conventions in the documentation for the naming of those nodes?

Reply:
Could do, although this is effectively user choice. In the end the names are purely labels for debugging. It could be useful to clarify that they're debug labels however.